docs/adr-011-extract-register-wasm-helpers.md

```
@@ -0,0 +1,115 @@
+<!-- markdownlint-disable-next-line MD013 -->
+# Architectural decision record (ADR) 011: Extract `register_wasm` helpers to reduce cyclomatic complexity
+
+# Architectural decision record (ADR) 011: Extract `register_wasm` helpers to reduce cyclomatic complexity
```

markdownlint-cli2 (0.22.1)

[warning] Multiple headings with the same content

(MD024, no-duplicate-heading)